com.unity.robotics.urdf-importer/Tests/Runtime/SensorTests.cs

```diff
@@ -0,0 +1,168 @@
+        Assert.AreEqual(totalData,sensor.elements.Count);
+    }
+
+    [Test]
+    public void NumberofAttributeTest()
+    {
+        int totalAttributes = 5;
+        int numberofAttributes = 0;
+        XDocument xdoc = XDocument.Load(GetSample());
+        Sensor sensor = new Sensor(xdoc.Element("sensor"));
+        foreach(var key in sensor.elements.Keys)
+        {
+            if(key.Contains("@"))
+            {
+                numberofAttributes++;
+            }
+        }
+        Assert.AreEqual(totalAttributes , numberofAttributes);
+    }
+
+    [Test]
+    public void NumberofElementsTest()
+    {
+        int totalElements = 8;
+        int numberofElements = 0;
+        XDocument xdoc = XDocument.Load(GetSample());
+        Sensor sensor = new Sensor(xdoc.Element("sensor"));
+        foreach(var key in sensor.elements.Keys)
+        {
+            if(!(key.Contains("@")))
+            {
+                numberofElements++;
+            }
+        }
+        Assert.AreEqual(totalElements , numberofElements);
+    }
+
+    [Test]
+    public void NoEmptyValue()
+    {
+        int emptyValue = 0;
+        XDocument xdoc = XDocument.Load(GetSample());
+        Sensor sensor = new Sensor(xdoc.Element("sensor"));
+        foreach(var value in sensor.elements.Values)
+        {
+            if(value == "" || value == null)
+            {
+                emptyValue++;
+            }
+        }
+        Assert.AreEqual(emptyValue , 0);       
+    }
+
+    [Test]
+    public void CorrectAttributeValues()
+    {
+        XDocument xdoc = XDocument.Load(GetSample());
+        XElement sensor = xdoc.Element("sensor");
+        Assert.IsNotNull(sensor);
+        XAttribute sensorName = sensor.Attribute("name");
+        Assert.IsNotNull(sensorName);
+        Assert.AreEqual(sensorName.Value,"my_camera");
+        XAttribute sensorType = sensor.Attribute("type");
+        Assert.IsNotNull(sensorType);
+        Assert.AreEqual(sensorType.Value,"camera");
+
+        XElement camera = sensor.Element("camera");
+        Assert.IsNotNull(camera);
+
+        XElement save = camera.Element("save");
+        Assert.IsNotNull(save);
+        XAttribute enabled = save.Attribute("enabled");
+        Assert.AreEqual(enabled.Value,"true");
+        XElement path = save.Element("path");
+        Assert.IsNotNull(path);
+        Assert.AreEqual(path.Value,"/tmp/camera_save_tutorial");
+
+        XElement horizontal_fov = camera.Element("horizontal_fov");
+        Assert.IsNotNull(horizontal_fov);
+        Assert.AreEqual(horizontal_fov.Value,"1.047");
+
+        XElement image = camera.Element("image");
+        Assert.IsNotNull(image);
+
+        XElement width = image.Element("width");
+        Assert.IsNotNull(width);
+        Assert.AreEqual(width.Value,"1920");
+
+        XElement height = image.Element("height");
+        Assert.IsNotNull(height);
+        Assert.AreEqual(height.Value,"1080"); 
+
+        XElement clip = camera.Element("clip");
```

Comment:
Since you have already done the parsing in Sensor.cs, here for an actual unit testing, you need to verify that the parsing was correct, by reading the already extracted sensor attributes from the Sensor.elements dictionary instead of re-parsing.
e.g.
``` 
Sensor sensor = ....
Assert.AreEqual(sensor.elements.get("sensor/camera/image/width", 1920)
```